In [6]:
import pandas as pd
movies_data = pd.read_csv("movies/movies.csv",  encoding='latin-1')

# Leer datos de un csv
movies_data

id    budget                            genres  \
0          5   4000000                      Crime|Comedy   
1          6  21000000             Action|Thriller|Crime   
2         11  11000000  Adventure|Action|Science Fiction   
3         12  94000000                  Animation|Family   
4         13  55000000              Comedy|Drama|Romance   
...      ...       ...                               ...   
9995  920081         0                     Action|Horror   
9996  920143         0                            Comedy   
9997  922017         0                            Comedy   
9998  922162         0                               NaN   
9999  922260    254056           Mystery|Thriller|Horror   

                                               homePage  \
0             https://www.miramax.com/movie/four-rooms/   
1                                                   NaN   
2     http://www.starwars.com/films/star-wars-episod...   
3                 http://movies.disney.com/finding-nemo   
4                                                   NaN   
...                                                 ...   
9995                                                NaN   
9996                                                NaN   
9997                                                NaN   
9998             https://www.netflix.com/title/81425229   
9999                                                NaN   

                                productionCompany productionCompanyCountry  \
0                            Miramax|A Band Apart                    US|US   
1      Universal Pictures|Largo Entertainment|JVC                 US|US|JP   
2                      Lucasfilm|20th Century Fox                    US|US   
3                                           Pixar                       US   
4               Paramount|The Steve Tisch Company                      US|   
...                                           ...                      ...   
9995                                          NaN                      NaN   
9996  Caracol Televisión|Dago García Producciones                    CO|CO   
9997                                          NaN                      NaN   
9998                                          NaN                      NaN   
9999          Central Partnership|KIT Film Studio                    RU|RU   

                   productionCountry      revenue  runtime  video  ...  \
0           United States of America    4257354.0       98  False  ...   
1     Japan|United States of America   12136938.0      110  False  ...   
2           United States of America  775398007.0      121    NaN  ...   
3           United States of America  940335536.0      100    NaN  ...   
4           United States of America  677387716.0      142  False  ...   
...                              ...          ...      ...    ...  ...   
9995                             NaN          0.0      100  False  ...   
9996                        Colombia          0.0       97  False  ...   
9997                         Nigeria          0.0      112  False  ...   
9998        United States of America          0.0       59  False  ...   
9999                          Russia     254042.0        0    NaN  ...   

     popularity releaseDate voteAvg voteCount genresAmount productionCoAmount  \
0        20.880  1995-12-09     5.7      2077            2                  2   
1         9.596  1993-10-15     6.5       223            3                  3   
2       100.003  1977-05-25     8.2     16598            3                  2   
3       134.435  2003-05-30     7.8     15928            2                  1   
4        58.751  1994-07-06     8.5     22045            3                  2   
...         ...         ...     ...       ...          ...                ...   
9995     16.662  2021-11-26     6.8       108            2                  1   
9996    491.706  2021-12-25     1.5         2            1                  2   
9997    565.658  2021-12-

# 1 Exploración rapida de los datos

## 1.1 Resumen del set de datos

In [7]:
movies_data.describe()

id        budget       revenue       runtime    popularity  \
count   10000.000000  1.000000e+04  1.000000e+04  10000.000000  10000.000000   
mean   249876.829300  1.855163e+07  5.673793e+07    100.268100     51.393907   
std    257380.109004  3.662669e+07  1.495854e+08     27.777829    216.729552   
min         5.000000  0.000000e+00  0.000000e+00      0.000000      4.258000   
25%     12286.500000  0.000000e+00  0.000000e+00     90.000000     14.577750   
50%    152558.000000  5.000000e+05  1.631245e+05    100.000000     21.905500   
75%    452021.750000  2.000000e+07  4.479661e+07    113.000000     40.654000   
max    922260.000000  3.800000e+08  2.847246e+09    750.000000  11474.647000   

            voteAvg     voteCount  genresAmount  productionCoAmount  \
count  10000.000000  10000.000000  10000.000000        10000.000000   
mean       6.483490   1342.381800      2.596500            3.171400   
std        0.984274   2564.196637      1.154565            2.539738   
min        1.300000      1.000000      0.000000            0.000000   
25%        5.900000    120.000000      2.000000            2.000000   
50%        6.500000    415.000000      3.000000            3.000000   
75%        7.200000   1316.000000      3.000000            4.000000   
max       10.000000  30788.000000     16.000000           89.000000   

       productionCountriesAmount   actorsAmount  
count               10000.000000   10000.000000  
mean                    1.751000    2147.666600  
std                     3.012093   37200.075802  
min                     0.000000       0.000000  
25%                     1.000000      13.000000  
50%                     1.000000      21.000000  
75%                     2.000000      36.000000  
max                   155.000000  919590.000000

## 1.2 Cuenta de valores nulos

In [8]:

null_count = movies_data.isnull().sum()
null_count

id                              0
budget                          0
genres                         53
homePage                     5807
productionCompany             457
productionCompanyCountry     1280
productionCountry             233
revenue                         0
runtime                         0
video                         486
director                       74
actors                         80
actorsPopularity               87
actorsCharacter                47
originalTitle                   0
title                           0
originalLanguage                0
popularity                      0
releaseDate                     0
voteAvg                         0
voteCount                       0
genresAmount                    0
productionCoAmount              0
productionCountriesAmount       0
actorsAmount                    0
castWomenAmount                 0
castMenAmount                   0
dtype: int64

## 1.3 El tipo de cada variable

In [9]:
movies_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10000 entries, 0 to 9999
Data columns (total 27 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   id                         10000 non-null  int64  
 1   budget                     10000 non-null  int64  
 2   genres                     9947 non-null   object 
 3   homePage                   4193 non-null   object 
 4   productionCompany          9543 non-null   object 
 5   productionCompanyCountry   8720 non-null   object 
 6   productionCountry          9767 non-null   object 
 7   revenue                    10000 non-null  float64
 8   runtime                    10000 non-null  int64  
 9   video                      9514 non-null   object 
 10  director                   9926 non-null   object 
 11  actors                     9920 non-null   object 
 12  actorsPopularity           9913 non-null   object 
 13  actorsCharacter            9953 non-null   obje

Aca podemos observar que mas del 50% de las peliculas no cuenta con una pagina de inicio o al menos no esta en el data set

# 2 Tipo de cada variable y descripción

|          Variable         | Tipo (Cuantitativa o   Cualitativa) | Discreta o Continua / Nominal o   Ordinal | Tipo del Dato |                        Ejemplo                       | descripción                                                       |
|:-------------------------:|:-----------------------------------:|:-----------------------------------------:|:-------------:|:----------------------------------------------------:|-------------------------------------------------------------------|
|             id            |             Cuantitativa            |                  Discreta                 |     int64     |                           1                          | Id de la película                                                 |
|           budget          |             Cuantitativa            |                  Continua                 |     int64     |                        1000000                       | El presupuesto para la película                                   |
|           genres          |             Cualitativa             |                  Nominal                  |     object    |                    Action, Comedy                    | El género de la película                                          |
|          homePage         |             Cualitativa             |                  Nominal                  |     object    |                    www.example.com                   | La página de inicio de la película                                |
|     productionCompany     |             Cualitativa             |                  Nominal                  |     object    |                 Miramax\|A Band Apart                | Las compañías productoras de la película.                         |
|  productionCompanyCountry |             Cualitativa             |                  Nominal                  |     object    |                      US\|US\|JP                      | Países de las compañías productoras de la película                |
|     productionCountry     |             Cualitativa             |                  Nominal                  |     object    |       United States of America\|Canada\|Germany      | Países en los que se llevó a cabo la producción  de la película   |
|          revenue          |             Cuantitativa            |                  Continua                 |    float64    |                       1500000.5                      | El ingreso de la película                                         |
|          runtime          |             Cuantitativa            |                  Continua                 |     int64     |                          120                         | La duración de la película                                        |
|           video           |             Cualitativa             |                  Nominal                  |     object    |                         FALSE                        | Si tiene videos promocionales o no                                |
|          director         |             Cualitativa             |                  Nominal                  |     object    |                   Christopher Nolan                  | Director de la película                                           |
|           actors          |             Cualitativa             |                  Nominal                  |     object    |                   Leonardo DiCaprio                  | Actores que participan en la película (Elenco)                    |
|      actorsPopularity     |             Cualitativa             |                  Nominal                  |     object    |           \| 9.008\|6.383\|10.757\|18.295\|          | Indice de popularidad del elenco de la película.                  |
|      actorsCharacter      |             Cualitativa             |                  Nominal                  |     object    |  Ted the   Bellhop\|Angela\|Man\|Athena\|Siegfried\| | Personaje que interpreta cada actor en la película                |
|       originalTitle       |             Cualitativa             |                  Nominal                  |     object    |                       Inception                      | El título original de la película, en su idioma original          |
|           title           |             Cualitativa             |                  Nominal                  |     object    |                       Inception                      | El título de la película traducido al inglés                      |
|      originalLanguage     |             Cualitativa             |                  Nominal                  |     object    |                        English                       | Idioma original en que se encuentra la película                   |
|         popularity        |             Cuantitativa            |                  Continua                 |    float64    |                         95.6                         | Índice de popularidad de la película calculado semanalmente       |
|        releaseDate        |             Cualitativa             |                  Nominal                  |     object    |                       7/16/2010                      | Fecha de lanzamiento de la película                               |
|          voteAvg          |             Cuantitativa            |                  Continua                 |    float64    |                          8.5                         | El promedio de los votos en la plataforma para la película        |
|         voteCount         |             Cuantitativa            |                  Discreta                 |     int64     |                         12000                        | El número de votos en la plataforma para la película              |
|        genresAmount       |             Cuantitativa            |                  Discreta                 |     int64     |                           3                          | Cantidad de géneros que representan la película                   |
|     productionCoAmount    |             Cuantitativa            |                  Discreta                 |     int64     |                           2                          | Cantidad de compañías productoras que participaron en la película |
| productionCountriesAmount |             Cuantitativa            |                  Discreta                 |     int64     |                           5                          | Cantidad de países en los que se rodó la película                 |
|        actorsAmount       |             Cuantitativa            |                  Discreta                 |     int64     |                          10                          | Cantidad de personas que actúan en la película                    |
|      castWomenAmount      |             Cualitativa             |                  Nominal                  |     object    |                           5                          | Cantidad de actrices en el elenco de la película                  |
|       castMenAmount       |             Cualitativa             |                  Nominal                  |     object    |                           8                          | Cantidad de actores en el elenco de la película                   |


# Preguntas

## a) ¿Cuáles son las 10 películas que contaron con más presupuesto?

In [16]:
movies_data[["budget","title"]].sort_values(by = "budget", ascending=False).head(10)

budget                                        title
716   380000000  Pirates of the Caribbean: On Stranger Tides
4710  365000000                      Avengers: Age of Ultron
5952  356000000                            Avengers: Endgame
4953  300000000                               Justice League
5953  300000000                       Avengers: Infinity War
...         ...                                          ...
6376          0                                    3 Idiotas
2637          0                           Kronk's New Groove
2638          0                              Avenging Angelo
2641          0                Barbie in 'A Christmas Carol'
1894          0           An American Tail: Fievel Goes West

[10000 rows x 2 columns]